### DISCUSSION TITLE
Unable to get the motion detection to work

### DISCUSSION THREAD

**Discussion by mzelem** [QuantumEntangledAndy/neolink#293] [Q&A]
I setup neolink to relay the video from a Reolink Argus Eco. Everything works great except if I add the on_motion option to the config I'm not able to see the video and I never see the line in the log about how the stream is available at x address. I see a warning in the log about a decode error from push notifications and I'm wondering if that is related?
The error is:
```
neolink::common::pushnoti] Issue connecting to push notifications server: Http(reqwest::Error { kind: Decode, source: Error("expected value", line: 1, column: 1) })
```

I'm using version: `v0.6.3.rc.2`

The config is:
```
bind = "0.0.0.0"

[[cameras]]
name = "Reolink1"
username = "admin"
password = "*****"
discovery = "local"
address = "****************"
uid = "********************"
stream = "Main"
idle_disconnect = true
  [cameras.pause]
  on_motion = true # Should pause when no motion
  on_client = true # Should pause when no rtsp client
  timeout = 2.1 # How long to wait after motion stops before pausing
```

I use it to connect the camera to Shinobi so the on_client option is not sufficient because there's basically always a client.

Thanks in advance for any help

**Comment by QuantumEntangledAndy** [comment]
Google removed the push notifications api we used. So we can't get motion detection while disconnected anymore

**Reply by mzelem** [reply]
What do you mean, FCM API changed, or we can't get the firebase tokens from the camera anymore?

I only ask because if it's fixable, maybe I can help. I don't have experience with Rust but I am a back-end developer.

**Reply by QuantumEntangledAndy** [reply]
On June 20, 2024, Google shut down the API call that was being used to get the token. (This was an undocumented api call since we were using it via reverse engineering)

In order to use the new api calls we would need the public vapid key which I have yet to be able to find by dumping the apk

**Reply by QuantumEntangledAndy** [reply]
There are some discussions about this here https://github.com/MatthieuLemoine/push-receiver/issues/68

**Reply by QuantumEntangledAndy** [reply]
What we try to do is to emulate being an android phone so that the push notifications will get sent to us.

To do this the the api calls that the phone sent out were captured and reverse engineered. Those calls were changed and now we aren't emulating being an android correctly anymore.

Id like to get a capture of an android re-negotiating a token and see how the calls have changed there. These calls are un documented by Google since they are private and part of the android client.

What can be done though is emulating the new vapid login method (designed for web browsers, and thus documented). However in order to use this I'd need to know the Reolink's public vapid key

**Reply by mzelem** [reply]
okay, I see. When I have some time I'll see what I can find.
